x-pack/plugins/security/public/account_management/user_profile/user_profile.tsx

```diff
@@ -137,6 +142,95 @@ function UserDetailsEditor({ user }: { user: AuthenticatedUser }) {
   );
 }
 
+function UserSettingsEditor({ formik }: { formik: ReturnType<typeof useUserProfileForm> }) {
+  if (!formik.values.data) {
+    return null;
+  }
+
+  return (
+    <EuiDescribedFormGroup
+      fullWidth
+      fieldFlexItemProps={{ style: { alignSelf: 'flex-start' } }}
+      title={
+        <h2>
+          <FormattedMessage
+            id="xpack.security.accountManagement.userProfile.userSettingsTitle"
+            defaultMessage="User Settings"
+          />
+        </h2>
+      }
+      description={
+        <FormattedMessage
+          id="xpack.security.accountManagement.userProfile.themeFormGroupDescription"
+          defaultMessage="Choose your desired theme for Kibana"
+        />
+      }
+    >
+      <EuiFlexGroup responsive={false}>
+        <EuiFlexItem>
+          <FormRow
+            name="data.userSettings.darkMode"
+            label={
+              <FormLabel for="data.userSettings.darkMode">
+                <FormattedMessage
+                  id="xpack.security.accountManagement.userProfile.userSettings.theme"
+                  defaultMessage="Theme"
+                />
+              </FormLabel>
+            }
+            fullWidth
+          >
+            <EuiButtonGroup
+              legend={i18n.translate(
+                'xpack.security.accountManagement.userProfile.userSettings.themeGroupDescription',
+                {
+                  defaultMessage: 'Kibana Theme',
+                }
+              )}
+              buttonSize="m"
+              data-test-subj="darkModeButton"
+              idSelected={formik.values.data.userSettings.darkMode}
+              options={[
```

Comment:
Was any thought given to the possibility of including an option that detects the user's current operating system or user agent settings to apply their desired theme automatically (ex. the "Match OS" option I show in the older designs)? Or is that a future enhancement?

![image](https://user-images.githubusercontent.com/3884767/234038087-e22f9a9b-db37-40cd-ba0c-9375bfb3b86d.png)

Reply:
Yes, I looked into adding that option, but based on how the metadata is provided from the server, it's not possible at this time (or at least it wasn't apparent to me!)

I would like to add this for a future release.